preupg/xml_to_json.py

```diff
@@ -0,0 +1,404 @@
+            all_results.update(rule_result_dict)
+
+        self.addOptToResultDict(self.result_dict, "results", all_results)
+
+    def parseDictOfElements(self, outer_element, elements_dict):
+        """
+            Parse text or attributes inside
+            outer_element into dictionary
+            based on element_dict
+
+        element_dict :
+           {result: text} -> text
+           {check: system} -> attribute system
+           {check: [system,text]} -> more than one attr/text
+                                    from the same element
+        """
+        element_dict = {}
+        for element_name, attribute in elements_dict.iteritems():
+            try:
+                element = self.findInnerElement(outer_element, element_name)
+            except MissingElementException as missing_ele_report:
+                log_message(missing_ele_report, level=logging.ERROR)
+
+            self.checkFoundElement(element, attribute)
+            if element:
+                try:
+                    element_dict.update(self.parseElementAttributes(element, attribute))
+                except EmptyTextException:
+                    pass
+                except AttrNotFoundException as err_mesg:
+                    log_message(err_mesg, level=logging.ERROR)
+
+        return element_dict
+
+    def checkFoundElement(self, element, attribute):
+        '''
+        Control each element whether it makes sense
+        to process them (like text is empty attribute
+        doesn't exit etc..)
+
+        '''
+        for elem in element:
+            # {check: [system,text]}
+            if type(attribute) is list:
+                for attr in attribute:
+                    try:
+                        self.checkForValidAttribute(elem, attr)
+                    except AttrNotFoundException as err_mesg:
+                        attribute.remove(attr)
+                        if self.verbose:
+                            log_message(err_mesg, level=logging.ERROR)
+            # {check: system} -> attribute system
+            else:
+                try:
+                    self.checkForValidAttribute(elem, attribute)
+                except AttrNotFoundException as err_mesg:
+                    element.remove(elem)
+                    if self.verbose:
+                        log_message(err_mesg, level=logging.ERROR)
+
+    def checkForValidAttribute(self, element, attribute):
+        '''
+        Check if text or attribute is empty or missing
+        if so make report about it and don't bother with it anymore
+        '''
+        if attribute == "text":
+            text = self.getDataFromElement(element, attribute)
+            if text is None or text.strip() == '':
+                raise AttrNotFoundException("Text in \"{0}\" element is empty doesn't exist"
+                                   .format(self.removeNamespace(element)))
```

Comment:
When I fix the verbose logging I'm getting huge number of exactly this sentence:
`Text in "check-import" element is empty doesn't exist`
It does not say much. I suggest:
```
raise AttrNotFoundException("Text in \"{0}\" element on line {1} is empty".format(self.removeNamespace(element), element.sourceline))
```
but for that you need to substitute
~~import xml.etree.ElementTree as ET~~
by 
`from lxml import etree as ET`

lxml is just improved ElementTree library (and it's available on both vanilla RHEL5 and RHEL6).